test/CadlRanchProjects.Tests/type-property-optionality.cs

```diff
@@ -590,5 +589,90 @@ public void PlainTime()
             var output = PlainTimeProperty.DeserializePlainTimeProperty(document.RootElement);
             Assert.AreEqual(PlainTimeData, output.Property);
         }
+
+        [Test]
+        public void NotRequiredNullablePropertiesOmitedByDefault()
+        {
+            var inputModel = new PlainTimeProperty();
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.False(element.TryGetProperty("property", out _));
+        }
+
+        [Test]
+        public void NotRequiredPropertiesDeserializedWithNullsWhenUndefined()
+        {
+            var model = PlainTimeProperty.DeserializePlainTimeProperty(JsonDocument.Parse("{}").RootElement);
+            Assert.Null(model.Property);
+        }
+
+        [Test]
+        public void NotRequiredPropertiesDeserializedWithValues()
```

Comment:
we should use `Optional` instead of `NotRequired`

Reply:
fixed